src/components/Questionnaire/QuestionnaireForm.tsx

```diff
@@ -532,6 +542,60 @@ export function QuestionnaireForm({
             firstErrorId = firstErrorId ? firstErrorId : q.id;
           }
         }
+
+        if (q.type === "structured") {
+          const response = form.responses.find((r) => r.question_id === q.id);
+
+          switch (q.structured_type) {
+            case "appointment": {
+              const appointmentData =
+                (response?.values?.[0]?.value as CreateAppointmentQuestion[]) ||
+                [];
+
+              const appointmentErrors = validateAppointmentQuestion(
+                appointmentData[0],
+                q.id,
+              );
+              errors.push(...appointmentErrors);
+              if (appointmentErrors.length > 0) {
+                firstErrorId = firstErrorId ? firstErrorId : q.id;
+              }
+              break;
+            }
+            case "medication_statement": {
+              const medicationData =
+                (response?.values?.[0]
+                  ?.value as MedicationStatementRequest[]) || [];
+
+              const medicationErrors = validateMedicationStatementQuestion(
+                medicationData,
+                q.id,
+              );
+              errors.push(...medicationErrors);
+              if (medicationErrors.length > 0) {
+                firstErrorId = firstErrorId ? firstErrorId : q.id;
+              }
+              break;
+            }
+            case "medication_request": {
+              const medicationData =
+                (response?.values?.[0]?.value as MedicationRequest[]) || [];
+
+              const medicationErrors = validateMedicationRequestQuestion(
+                medicationData,
+                q.id,
+                t,
+              );
+              errors.push(...medicationErrors);
+              if (medicationErrors.length > 0) {
+                firstErrorId = firstErrorId ? firstErrorId : q.id;
+              }
+              break;
+            }
```

Comment:
not a huge fan of this abstract/generic component performing validations of very specific structured questions. could end up being hard to maintain later on.

cc: @bodhish

Reply:
thought:


We could create a map of (keys being structured question's key; value's being a callback fn.) and make this generic component call all available extra validations here present in that dict.